src/main/java/apoc/export/cypher/formatter/AbstractCypherFormatter.java

```diff
@@ -127,18 +129,23 @@ public void buildStatementForNodes(String nodeClause, String setClause,
 				// properties
 				out.append(", ");
 				out.append("properties:");
-				writeProperties(out, props, (es) -> !idMap.containsKey(es.getKey()));
+
 				propertiesCount.addAndGet(props.size());
+				props = props.entrySet()
+						.stream()
+						.filter(es -> !idMap.containsKey(es.getKey()))
+						.collect(Collectors.toMap(es -> es.getKey(), es -> es.getValue()));
+				writeProperties(out, props);
 
 				// end element
 				out.append("}");
 				boolean isEnd = last.equals(node);
 				if (isEnd || batchCount.get() % unwindBatchSize == 0 || batchCount.get() % batchSize == 0) {
```

Comment:
the complexity of this still very high

why can't we combine unwindBatchSize/batchSize e.g. via   minimum of both?